common/unsafe/src/main/java/org/apache/spark/unsafe/UnsafeAlignedOffset.java

```diff
@@ -28,12 +28,20 @@
 
   private static final int UAO_SIZE = Platform.unaligned() ? 4 : 8;
 
+  private static int TEST_UAO_SIZE = 0;
+
+  // used for test only
+  public static void setUaoSize(int size) {
+    assert size == 0 || size == 4 || size == 8;
+    TEST_UAO_SIZE = size;
+  }
+
   public static int getUaoSize() {
-    return UAO_SIZE;
+    return TEST_UAO_SIZE == 0 ? UAO_SIZE : TEST_UAO_SIZE;
```

Comment:
nit: you can use `Utils.isTesting` instead

Reply:
Other tests might not set UAO size manually, then we'll get 0 in this case.

Reply:
We should handle the logic of setting/reverting the size in the test code.

Reply:
then, we need to figure out all the test cases where used `UnsafeAlignedOffset`, right?

Reply:
Why do we need to do that? The default value of TEST_UAO_SIZE should be the same as UAO_SIZE, only if you want to test other values then you need to change it.

Reply:
Ok, I get your point. Let me update it later.

Reply:
hmm...unfortunately, `unsafe` project doesn't depend on `core`. So we can not access the `Utils`.